scripts/stitch.py

```diff
@@ -1,3692 +1,493 @@
+                    zoomed.paste(
+                        sprite,
+                        (
+                            mapsquare_x + int(round(256 * scaling_factor)),
+                            int(round(mapsquare_y + 256 * scaling_factor)),
+                        ),
+                        sprite,
+                    )
+
+            low_zoomed_x = int((map_low_x - 1) * scaling_factor + 0.01)
+            high_zoomed_x = int((map_high_x + 0.9 + 1) * scaling_factor + 0.01)
+            low_zoomed_y = int((map_low_y - 1) * scaling_factor + 0.01)
+            high_zoomed_y = int((map_high_y + 0.9 + 1) * scaling_factor + 0.01)
+            for x in range(low_zoomed_x, high_zoomed_x + 1):
+                for y in range(low_zoomed_y, high_zoomed_y + 1):
+                    coord_x = int((x - (map_low_x - 1) * scaling_factor) * 256)
+                    coord_y = int((y - (map_low_y - 1) * scaling_factor) * 256)
+                    cropped = zoomed.crop(
+                        (coord_x, zoomed.size[1] - coord_y - 256, coord_x + 256, zoomed.size[1] - coord_y)
+                    )
+
+                    if not all_black(cropped):
+                        out_path = os.path.join(out_tiles_dir, f"{map_id}/{zoom}/{plane}_{x}_{y}.png")
+                        mkdir_p(out_path)
+                        cropped.save(out_path)
+
+
+def main(version, select_maps=()):
+    version_dir = f"./out/mapgen/versions/{version}"
+    base_tiles_dir = os.path.join(version_dir, "tiles", "base")
+    out_tiles_dir = os.path.join(version_dir, "output", "tiles", "rendered")
+    icons_dir = os.path.join(version_dir, "output", "icons")
+
+    cache_defs_path = os.path.join(version_dir, "worldMapDefinitions.json")
+    extra_defs_path = os.path.join("scripts", "user_world_defs.json")
+    icons_path = os.path.join(version_dir, "minimapIcons.json")
+    basemaps_path = os.path.join(version_dir, "output", "basemaps.json")
+
+    defs = load_defs(cache_defs_path, extra_defs_path, base_tiles_dir)
+    icons = load_icons(icons_path)
+    icon_sprites = load_sprites(icons_dir)
+
+    basemaps = []
+    for map_id, defn in defs.items():
+        if select_maps and map_id not in select_maps:
+            continue
+        basemap = load_basemap(defn)
+        basemaps.append(basemap)
+        render_map(map_id, defn, icons, icon_sprites, base_tiles_dir, out_tiles_dir)
+
+    with open(basemaps_path, "w") as f:
+        json.dump(basemaps, f)
+
+
+if __name__ == "__main__":
+    # 1. download cache using cache.py (files go to "./data")
+    # 2. export base tiles using runelite MapExport.java (images go to "./out/mapgen/versions/{cache_date}/tiles/base")
+    # 3. run this file to render tiles (images go to "./out/mapgen/versions/{cache_date}/output/tiles/rendered")
+    # 4. zip rendered tiles + basemaps.json + map icons
+    # 5. upload zip to map server
+    # 6. update mapids page on wiki if any changes occurred
+    main("2024-07-24_a", (17, 34))
```

Comment:
Could you make this date a param or automatically generated per run?

Reply:
Probably not the smartest thing but I put the version name that cache.py generates into version.txt so that MapExport.java and stitch.py can grab it from there and do what they need to do. So now none of these files need any CLI arguments or anything.